pkg/cli/verifier/pod.go

```diff
@@ -0,0 +1,72 @@
+package verifier
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openservicemesh/osm/pkg/constants"
+)
+
+// PodStatusVerifier implements the Verifier interface for control plane health
+type PodStatusVerifier struct {
+	stdout     io.Writer
+	stderr     io.Writer
+	kubeClient kubernetes.Interface
+	app        types.NamespacedName
```

Comment:
It seems a bit unintuitive to me to have the `.Name` of this correspond to the `app` label instead of the pod's actual name. Could this be a plain label selector instead, or even a plain list of pods that's constructed elsewhere? I think that would be clearer and more widely applicable.

Reply:
The idea was to not use pod names here, as there may be many of them. The tool's job is to figure out the pods (which it can) and determine whether they are running. I didn't want to expose additional labels as they wouldn't be used. But the variable is called `app`, so I personally didn't find this ambiguous. I also used the NamespacedName type to avoid passing an additional namespace argument to the inner functions.